plugins/modules/template_workflow_manager.py

```diff
@@ -2413,24 +3086,126 @@ def get_diff_merged(self, config):
+                        self.log("Successfully collected the tag id {0} for the tag {1}".format(tag_id, tag_name), "INFO")
+                        # Get the device ids associated with the given tag for given site
+                        tag_device_ids = self.get_device_ids_from_tag(tag_id, tag_name)
+                        self.log("Successfully collected the device ids {0} associated with the tag {1}".format(tag_device_ids, tag_name), "INFO")
+
+                    self.log("Getting the device ids based on device assoicated with tag or site or both.", "DEBUG")
+
+                    if tag_device_ids and site_assign_device_ids:
+                        self.log("Getting the common device ids based on devices fetched from site and with tag.", "DEBUG")
+                        common_device_ids = list(set(tag_device_ids).intersection(set(site_assign_device_ids)))
+                        device_ids.extend(common_device_ids)
+                    elif site_assign_device_ids and not tag_device_ids:
+                        self.log("Getting the device ids based on devices fetched from site.", "DEBUG")
+                        device_ids.extend(site_assign_device_ids)
+                    elif tag_device_ids and not site_assign_device_ids:
+                        self.log("Getting the device ids based on devices fetched with the tag {0}.".format(tag_name), "DEBUG")
+                        device_ids.extend(tag_device_ids)
+                    else:
+                        self.log(
+                            "There is no matching device ids found for the deployment of template {0} "
+                            "for the given site {1}".format(template_name, site_name), "WARNING"
+                        )
+                        continue
+
+                device_missing_msg = (
+                    "There is no device id found for the given site(s) '{0}' in the "
+                    "Cisco Catalyst Center so cannot deploy the template '{1}'."
+                ).format(site_name_list, template_name)
+            else:
+                self.msg = (
+                    "Unable to provision the template '{0}' as device related details are "
+                    "not given in the playboook. Please provide it either via the parameter "
+                    "device_specific_details or with site_associated_provisioning."
+                ).format(self.msg)
+                self.set_operation_result("failed", False, self.msg, "INFO").check_return_status()
+
+            if not device_ids:
+                self.msg = device_missing_msg
+                self.set_operation_result("failed", False, self.msg, "INFO")
                 return self
 
+            device_ips = self.get_device_ips_from_device_ids(device_ids)
+            self.log("Successfully collect the device ips {0} for the device ids {1}.".format(device_ips, device_ids), "INFO")
+            deploy_temp_payload = self.create_payload_for_template_deploy(deploy_temp_details, device_ids)
```

Comment:
self.log("Deployment payload created successfully for template '{0}'.".format(template_name), "INFO")